### PR TITLE
Makeshift pistol

### PR DESCRIPTION
**Pull request by Bucovineanu**
## About The Pull Request
image
Adds a makeshift pistol (requires a slide (made at a hacked autolathe), a receiver, a pipe and wrapping, as well as a saw and a screwdriver as tools) and magazine (requires wirecutters for uh.. bending, uses metal sheets and a mag spring (also made at a hacked autolathe) that can be made out of several spare parts. It acts like a lower capacity version of the stetchkin that randomly jams after firing, requiring you to manually move the slide. The sprite is roughly based off of the hipoint C9

## Why It's Good For The Game
It gives the "modular" receiver more than one use (improv shotgun), making its name more fitting, and gives the crew another option for a (makeshift) ballistic gun that takes some effort to make. Due to its unreliability, low capacity and inability to take makarov magazines it's a very situational weapon, which I think should position itself somewhere around where the improv shotgun is (which mind you, can be fed with scatter laser shells). 

## Changelog
:cl:
add: Added the makeshiftov pistol and its respective magazine
add: Added magazine spring, pistol slide to the hacked autolathe
imageadd: Added some icons for the makeshift pistol, its magazine and the parts required to make it.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/boxes_magazines/external/pistol.dm [hunk 1]
```diff
@@ -83,3 +83,16 @@
...
+	icon_state = "9x19mks-6"
+	desc= "A makeshift pistol mag, made out of spare parts. Doesn't look too reliable."
+	ammo_type = /obj/item/ammo_casing/c9mm
+	caliber = "9mm"
+	max_ammo = 6
+	start_empty = TRUE
+
+/obj/item/ammo_box/magazine/m9mm_mkshft/update_icon()
+	..()
+	icon_state = "9x19mks-[ammo_count() ? "6" : "0"]"
```
Comment (necromanceranne) [inline code/modules/projectiles/boxes_magazines/external/pistol.dm:98]: Newline or travis will spill milk on your shoes.

File: code/datums/components/crafting/guncrafting.dm [hunk 2]
```diff
@@ -14,3 +14,15 @@
...
+	name = "pistol slide"
+	desc = "A striker-fired pistol slide with mounting brackets for safeties, decockers and other advanced gun features."
+	icon = 'icons/obj/improvised.dmi'
+	icon_state = "slide"
+
+/obj/item/weaponcrafting/magspring
+	name = "magazine spring"
+	desc = "A magazine spring intended as a spare part for old firearms."
+	icon = 'icons/obj/improvised.dmi'
+	icon_state = "magspring"
```
Comment (necromanceranne) [inline code/datums/components/crafting/guncrafting.dm:28]: New line here too.

File: code/modules/projectiles/boxes_magazines/external/pistol.dm (hunk 1 above)
Comment (Bucovineanu) [reply code/modules/projectiles/boxes_magazines/external/pistol.dm]: Yeah fixed that just now

**Comment by Pugie** [comment]
7 round pistol that does 30 brute and has mass printable ammo yikes.
I think "powercreep" is more apt a term than "situational"

**Comment by Tharcoonvagh** [comment]
Lower the damage output and add an at least 30% chance that it shoots you in the fucking head, it is a scavenged thing after all
Also id make it look more like the pipe gun from fo4

**Comment by Tharcoonvagh** [comment]
Also give it a special ammo that can only be crafted

**Comment by Bucovineanu** [comment]
Right now, it jams every 2 or 3 rounds. Even with 30 brute that's 90 damage before you need to start babysitting it, though I see the point in that unjamming doesn't take *that* long. I suppose a lower damage, separate ammo would be better, maybe 20 damage?

**Comment by necromanceranne** [comment]
The weapon requires you to be able to access a protolathe to even make, which makes me really wonder if that actually qualifies as an improvised weapon in the first place (the improvised shotgun is as much guilty of this as well). 

Additionally, if the only way to make it balanced would be making it a trap to use in the first place, you might as well not make it in the first place.

**Comment by Bucovineanu** [comment]
It requires access to an autolathe, not a protolathe, same as the improvised shotgun.
Also, I tried testing dual wielding as it currently is (with 9mm, 30 damage) and it doesn't help much. I think I was overestimating it with "2-3 shots before it jams"; out of 5 or so attempts I've made so far, by dual wielding I've only had either gun not jam so that it can fire 4 bullets once. That then means paying attention to the sound the gun is making as you rack the slide (I think I missed it in the OP, there is a dry fire sound on failure to feed) on both guns while trying to shoot someone. It's not very practical while also trying to shoot someone.

**Comment by anconfuzedrock** [comment]
much much better than the mosin and in some ways better than auto rifles, the only non shotgun ballistics you can purchase.

**Comment by MrDoomBringer** [comment]
things we don't need
more improvised weapons
image

**Comment by Bucovineanu** [comment]
> much much better than the mosin and in some ways better than auto rifles, the only non shotgun ballistics you can purchase.

But it's really not, you fiddle just slightly less than the Mosin, and the Mosin does 60 damage. The autorifle does slightly less damage, but not only is it far higher capacity, it also doesn't require you to fuck around with the gun at all to fire it.

**Comment by Doosteroni** [comment]
Gross it looks like a hi-point

**Comment by optimumtact** [comment]
We really don't need improvised weapons

**Comment by optimumtact** [comment]
In fact we really don't need anymore weapons at all.

**Comment by MrDoomBringer** [comment]
copypasted sentiment from the last 3 times this PR has been made:
imo if you want to add new ways for assistants to kill people you should encourage people to murder with the items that already exist on the station. Slipping a sec officer and slamming an airlock closed on their head or stabbing their eye out with a pen is way more interesting than "unga me craft pistol"

**Pull request closed without merging**